### ISSUE TITLE
Support backpressure

### ISSUE THREAD

**Issue by feross** [feross/simple-peer#18]
Right now, we just send as fast as possible. We should support backpressure and detect when `bufferedAmount` gets too high and pause for a while.

Relevant:
- https://code.google.com/p/webrtc/issues/detail?id=3984
- https://github.com/mafintosh/webcat/issues/7

The solution implemented for `simple-peer` can also be used for websocket libraries like [`websocket-stream`](https://www.npmjs.com/package/websocket-stream), since data channels and websockets have the same API, and thus, the same difficulties detecting when backpressure is needed.

cc @mafintosh, @maxogden

**Comment by mafintosh** [comment]
I think an easy impl would be to poll `bufferedAmount` every 250 ms and only call the write callback if it is `0`. We should use a higher than normal watermark (like 1mb) since our feedback loop has higher latency because of the poll.

Something like this:

``` js
var SimplePeer = function () {
  stream.Duplex.call({highWaterMark: 1024 * 1024})
  ...
  var self = this
  setInterval(function () {
    if (self._datachannel.bufferedAmount || !self._cb) return
    var cb = self._cb
    self._cb = null
    cb()
  }, 250)
  ...
}
SimplePeer.prototype._write = function (data, enc, cb) {
  this._datachannel.send(data)
  if (!this._datachannel.bufferedAmount) return cb()
  this._cb = cb
}
```

It would also be interesting to investigate if there was a hack to detect when `bufferedAmount` changes (make it observable / use setters/getters)

**Comment by mafintosh** [comment]
Using the 1mb watermark and polling every 250ms would support 4mb/s which seems ok for now (we could dynamically adjust the watermark based how quickly the stream drains etc)

**Comment by feross** [comment]
@mafintosh Nice strategy. Just released it as 5.1.0.
